sympy/ntheory/generate.py

```diff
@@ -14,11 +14,16 @@
 from sympy.core.compatibility import as_int, range
 
 
+def _azeros(n):
```

Comment:
I wanted to get away from creating the `_array` in more than one place. The name ('l' or 'i') has to match. By localizing this in `_azeros` there is only one place that the array is created. But since these functions are now in one place it is easy to see that they all do the right thing so I am ok with it as is unless you think my former idea had merit.

Reply:
Yes, it looks clean. Fine.